tool/planet/etcd.go

```diff
@@ -412,23 +504,100 @@ func getBaseEtcdDir(version string) string {
 }
 
 func etcdRestore(file string) error {
-	ctx := context.TODO()
-	log.Info("Restoring backup to temporary etcd")
+	log.Info("Initializing new etcd database")
+
+	datadir, err := getEtcdDataDir()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// To be re-entrant, shutdown the etcd server if it's already running
+	err = etcdDisable(true, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// For the restore operation to be re-entrant, delete the data directory in-case the step is being re-run
+	err = os.RemoveAll(datadir)
```

Comment:
Should it be backed up first?

Reply:
I don't think so. All data within this directory, would be the result of a previous attempt to run this phase of the upgrade. If it failed, the data is incorrect, and if it succeeded, rerunning the step should produce the same output. 

I guess the only place where the data could be useful and different, is if someone got past the etcd upgrade, and force re-run this particular phase, it would then be missing later writes. However, if the cluster is rolled back prior to this step, the data will also be cleaned up, as the rollback goes back to the previous cluster state.